homeassistant/components/smarty/sensor.py

```diff
@@ -0,0 +1,237 @@
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.supply_air_temperature
+
+
+class ExtractAirTemperatureSensor(SmartySensor):
+    """Extract Air Temperature Sensor."""
+
+    def __init__(self, name, smarty):
+        """Supply Air Temperature Init."""
+        super().__init__(name=name + ' Extract Air Temperature',
+                         device_class=DEVICE_CLASS_TEMPERATURE,
+                         unit_of_measurement=TEMP_CELSIUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.extract_air_temperature
+
+
+class OutdoorAirTemperatureSensor(SmartySensor):
+    """Extract Air Temperature Sensor."""
+
+    def __init__(self, name, smarty):
+        """Outdoor Air Temperature Init."""
+        super().__init__(name=name + ' Outdoor Air Temperature',
+                         device_class=DEVICE_CLASS_TEMPERATURE,
+                         unit_of_measurement=TEMP_CELSIUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.outdoor_air_temperature
+
+
+class SupplyFanSpeedSensor(SmartySensor):
+    """Supply Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Supply Fan Speed RPM Init."""
+        super().__init__(name=name + ' Supply Fan Speed',
+                         device_class=DEVICE_CLASS_RPM,
+                         unit_of_measurement=RPM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.supply_fan_speed
+
+
+class ExtractFanSpeedSensor(SmartySensor):
+    """Extract Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Extract Fan Speed RPM Init."""
+        super().__init__(name=name + ' Extract Fan Speed',
+                         device_class=DEVICE_CLASS_RPM,
+                         unit_of_measurement=RPM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.extract_fan_speed
+
+
+class SystemStateSensor(SmartySensor):
+    """System State Sensor."""
+
+    def __init__(self, name, smarty):
+        """Alarm Sensor Init."""
+        super().__init__(name=name + ' System State',
+                         device_class=DEVICE_CLASS_STATUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.system_state
```

Comment:
What state is returned here?

Reply:
The system state shows additional information about which state the system is in.
Most important one is 'Boost', which is like the 4th speed and it's on for a period of time.

Stand-by, Building protection, Economy, Comfort, Emergency run, Preparing, Opening dampers, Boost, Cooling heaters, Closing dampers, Night Cooling, Critical alarm, Fire alarm, Heat exchanger frost protection, Change filters, Room RH 3 days average is lower than 30%. Limiting speed., DX cooler defrosting, Fire damper testing

Reply:
Some of these sound more like binary sensor types, eg "Fire alarm", "Critical alarm", "Room RH 3 days average is lower than 30%" and possibly others. Please review all of these and move appropriately to binary sensor entity platform.

Reply:
ok, I'll remove that sensor and create a new one for the boost which I think is most relevant.